### ISSUE TITLE
Question about TensorRT inference accuracy (using float16 datatype PyTorch model converted to onnx file) and unsupported UINT8 data type for Tensorflow model, trained on COCO dataset

### ISSUE THREAD

**Issue by phquanta** [NVIDIA/TensorRT#2047]
I used **COCO** dataset with 91 classes with additional class to train model using **pyTorch**. Then I converted the pyTorch model into an **onnx** file and run inference with **DetectNet inference using detectnet.cpp build** and it works in accurately detecting objects on detectnet. Then I used the same onnx file in the TensorRT inference that is in Isaac SDK and I don’t get the accurate inference and object detection as I get a lot of false positives for objects, i.e. when there is nothing in the camera feed view it says it detected something. Can you please take a look at this json file for the TensorRT inference and let me know of your suggestions to improve the accuracy of the confidence score and detection of objects? 

**Environment** 

**TensorRT Version:** 7.1.3.0
**NVIDIA GPU:** NVIDIA Jetson AGX Xavier
**CUDA Version:** 10.2
CUDNN Version:
**Operating System:** Ubuntu 18.04
Python Version (if applicable): 3.7
Tensorflow Version (if applicable):
PyTorch Version (if applicable):
Baremetal or Container (if so, version):
**torch**                              1.6.0
**torchvision**                        0.7.0

**ISAAC SDK json file for the TensorRT inference:**
{
“modules”: [
“detect_net”,
“ml”,
“perception”,
“sight”,
“viewers”
],
“graph”: {
“nodes”: [
{
“name”: “subgraph”,
“components”: [
{
“name”: “message_ledger”,
“type”: “isaac::alice::MessageLedger”
},
{
“name”: “interface”,
“type”: “isaac::alice::Subgraph”
}
]
},
{
“name”: “tensor_encoder”,
“components”: [
{
“name”: “message_ledger”,
“type”: “isaac::alice::MessageLedger”
},
{
“name”: “isaac.ml.ColorCameraEncoderCuda”,
“type”: “isaac::ml::ColorCameraEncoderCuda”
}
]
},
{
“name”: “tensor_r_t_inference”,
“components”: [
{
“name”: “message_ledger”,
“type”: “isaac::alice::MessageLedger”
},
{
“name”: “isaac.ml.TensorRTInference”,
“type”: “isaac::ml::TensorRTInference”
}
]
},
{
“name”: “detection_decoder”,
“components”: [
{
“name”: “message_ledger”,
“type”: “isaac::alice::MessageLedger”
},
{
“name”: “isaac.detect_net.DetectNetDecoder”,
“type”: “isaac::detect_net::DetectNetDecoder”
}
]
},
{
“name”: “detection_viewer”,
“components”: [
{
“name”: “isaac.alice.MessageLedger”,
“type”: “isaac::alice::MessageLedger”
},
{
“name”: “isaac.viewers.DetectionsViewer”,
“type”: “isaac::viewers::DetectionsViewer”
}
]
},
{
“name”: “color_camera_visualizer”,
“components”: [
{
“name”: “message_ledger”,
“type”: “isaac::alice::MessageLedger”
},
{
“name”: “isaac.viewers.ImageViewer”,
“type”: “isaac::viewers::ImageViewer”
}
]
},
{
“name”: “sight_widgets”,
“components”: [
{
“type”: “isaac::sight::SightWidget”,
“name”: “Detections”
}
]
}
],
“edges”: [
{
“source”: “subgraph/interface/image”,
“target”: “tensor_encoder/isaac.ml.ColorCameraEncoderCuda/rgb_image”
},
{
“source”: “tensor_encoder/isaac.ml.ColorCameraEncoderCuda/tensor”,
“target”: “tensor_r_t_inference/isaac.ml.TensorRTInference/image”
},
{
“source”: “tensor_r_t_inference/isaac.ml.TensorRTInference/bounding_boxes_tensor”,
“target”: “detection_decoder/isaac.detect_net.DetectNetDecoder/bounding_boxes_tensor”
},
{
“source”: “tensor_r_t_inference/isaac.ml.TensorRTInference/confidence_tensor”,
“target”: “detection_decoder/isaac.detect_net.DetectNetDecoder/confidence_tensor”
},
{
“source”: “detection_decoder/isaac.detect_net.DetectNetDecoder/detections”,
“target”: “detection_viewer/isaac.viewers.DetectionsViewer/detections”
},
{
“source”: “subgraph/interface/image”,
“target”: “color_camera_visualizer/isaac.viewers.ImageViewer/image”
},
{
“source”: “detection_decoder/isaac.detect_net.DetectNetDecoder/detections”,
“target”: “subgraph/interface/detections”
}
]
},
“config”: {
“color_camera_visualizer”: {
“isaac.viewers.ImageViewer”: {
“camera_name”: “camera”
}
},
“tensor_encoder”: {
“isaac.ml.ColorCameraEncoderCuda”: {
“rows”: 300,
“cols”: 300,
“pixel_normalization_mode”: “Unit”,
“tensor_index_order”: “201”
}
},
“tensor_r_t_inference”: {
“isaac.ml.TensorRTInference”: {
“model_file_path”: “/home/xavier/object_detection/coco_train/ssd-mobilenet.onnx”,
“engine_file_path”:"/home/xavier/object_detection/coco_train/ssd-mobilenet.plan",
“max_workspace_size”: 67108864,
“max_batch_size”: 32,
“inference_mode”: “float16”,
“force_engine_update”: false,
“input_tensor_info”: [
{
“operation_name”: “input_0”,
“channel”: “image”,
“dims”: [
3,
300,
300
],
“uff_input_order”: “channels_last”
}
],
“output_tensor_info”: [
{
“operation_name”: “boxes”,
“channel”: “bounding_boxes_tensor”,
“dims”: [
375,
32,
1
]
},
{
“operation_name”: “scores”,
“channel”: “confidence_tensor”,
“dims”: [
92,
3000,
1
]
}
]
}
},
“detection_decoder”: {
“isaac.detect_net.DetectNetDecoder”: {
“labels”: [
“BACKGROUND”,
“person”,
“bicycle”,
“car”,
“motorcycle”,
“airplane”,
“bus”,
“train”,
“truck”,
“boat”,
“traffic light”,
“fire hydrant”,
“street sign”,
“stop sign”,
“parking meter”,
“bench”,
“bird”,
“cat”,
“dog”,
“horse”,
“sheep”,
“cow”,
“elephant”,
“bear”,
“zebra”,
“giraffe”,
“hat”,
“backpack”,
“umbrella”,
“shoe”,
“eye glasses”,
“handbag”,
“tie”,
“suitcase”,
“frisbee”,
“skis”,
“snowboard”,
“sports ball”,
“kite”,
“baseball bat”,
“baseball glove”,
“skateboard”,
“surfboard”,
“tennis racket”,
“bottle”,
“plate”,
“wine glass”,
“cup”,
“fork”,
“knife”,
“spoon”,
“bowl”,
“banana”,
“apple”,
“sandwich”,
“orange”,
“broccoli”,
“carrot”,
“hot dog”,
“pizza”,
“donut”,
“cake”,
“chair”,
“couch”,
“potted plant”,
“bed”,
“mirror”,
“dining table”,
“window”,
“desk”,
“toilet”,
“door”,
“tv”,
“laptop”,
“mouse”,
“remote”,
“keyboard”,
“cell phone”,
“microwave”,
“oven”,
“toaster”,
“sink”,
“refrigerator”,
“blender”,
“book”,
“clock”,
“vase”,
“scissors”,
“teddy bear”,
“hair drier”,
“toothbrush”,
“hair brush”
],
“non_maximum_suppression_threshold”: 0.4,
“confidence_threshold”: 0.9,
“output_scale”: [720, 1280]
}
},
“sight_widgets”: {
“Detections”: {
“type”: “2d”,
“channels”: [
{ “name”: “$(fullname color_camera_visualizer/isaac.viewers.ImageViewer/image)” },
{ “name”: “$(fullname detection_viewer/isaac.viewers.DetectionsViewer/detections)” }
]
}
}
}
}

**Comment by nvpohanh** [comment]
@phquanta Could you try a newer TRT version (like TRT 8.0 or 8.2) and see if the accuracy issue still exists? If it does, please provide detailed steps to repro the issue. Thanks

**Comment by phquanta** [comment]
What version of TensorRT is used by the TensorRT inference in Isaac SDK?
[](https://docs.nvidia.com/isaac/doc/doc/component_api.html#isaac-ml-tensorrtinference)

**Comment by phquanta** [comment]
https://docs.nvidia.com/isaac/doc/doc/component_api.html#isaac-ml-tensorrtinference

**Comment by phquanta** [comment]
I am in process of trying the TRT 8.0 but was not clear about the version of TRT used by Isaac SDK.

**Comment by phquanta** [comment]
The issue was not with TensorRT inference as it works from Jetson inference on Xavier with version 7..1.3 but with the Decoder in Isaac SDK.  https://docs.nvidia.com/isaac/doc/doc/component_api.html#isaac-detect-net-detectnetdecoder

**Comment by nvpohanh** [comment]
I am not sure which version is used in ISAAC...

**Comment by ttyio** [comment]
closing due to no activity for more than 3 weeks, please reopen if you still have question, thanks!
